packages/person/src/person-avatar/element.ts

```diff
@@ -0,0 +1,173 @@
+import { CSSResult, TemplateResult, PropertyValues, html } from 'lit';
+import { property } from 'lit/decorators';
+import { ifDefined } from 'lit/directives/if-defined';
+import { PersonElement, PersonElementProps } from '../person';
+import { Availability, AccountType } from '../types';
+import Badge, { BadgeColor, IconName } from '@equinor/fusion-wc-badge';
+import Avatar, { AvatarSize } from '@equinor/fusion-wc-avatar';
+import style from './element.css';
+
+// persist elements
+Badge;
+Avatar;
+
+export type PersonAvatarElementProps = PersonElementProps & {
+  name?: string;
+  accountType?: AccountType;
+  availability?: Availability;
+  pictureSrc?: string;
+  size?: AvatarSize;
+  clickable?: boolean;
+  disabled?: boolean;
+  resolveDetails?: boolean;
+  resolvePresence?: boolean;
+  resolvePicture?: boolean;
+};
+
+export class PersonAvatarElement extends PersonElement implements PersonAvatarElementProps {
+  static styles: CSSResult[] = [style];
+
+  @property({ type: Boolean })
+  resolveDetails?: boolean = true;
+
+  @property({ type: Boolean })
+  resolvePresence?: boolean = true;
+
+  @property({ type: Boolean })
+  resolvePicture?: boolean = true;
+
+  @property({ type: String, reflect: true })
+  name?: string;
+
+  @property({ type: String, reflect: true })
+  accountType?: AccountType;
+
+  @property({ type: String, reflect: true })
+  availability?: Availability;
+
+  @property({ type: String, reflect: true })
+  pictureSrc?: string;
+
+  @property({ type: String, reflect: true })
+  size: AvatarSize = 'medium';
+
+  @property({ type: Boolean, reflect: true })
+  clickable?: boolean;
+
+  @property({ type: Boolean })
+  disabled?: boolean;
+
+  protected updated(changedProperties: PropertyValues) {
+    super.updated(changedProperties);
+    if (changedProperties.has('personResolver')) {
+      if (this.resolveDetails) {
+        this.resolveDetailsAsync();
+      }
+      if (this.resolvePresence) {
+        this.resolvePresenceAsync();
+      }
+      if (this.resolvePicture) {
+        this.resolvePictureAsync();
+      }
+    }
+  }
+
+  protected async resolveDetailsAsync() {
+    if (!this.name || !this.accountType) {
```

Comment:
i think the resolver should be the 'source' of truth, override with result.
like if the azure id changes, all data shoudl change too